### DISCUSSION TITLE
Crystal Plasticity- Deformation gradient error

### DISCUSSION THREAD

**Discussion by Emmanuel-324** [idaholab/moose#27263] [Q&A Getting Started]
Below is the error message

[31m
*** ERROR ***
The following error occurred in the object "MOOSE Problem", of type "FEProblem".

Material property 'deformation_gradient', requested by 'crysp_phase0' is not defined on block 0
Material property 'crysrot', requested by 'crysp_phase0' is not defined on block 0
Material property 'deformation_gradient', requested by 'crysp_phase1' is not defined on block 0
Material property 'crysrot', requested by 'crysp_phase1' is not defined on block 0
Material property 'deformation_gradient', requested by 'crysp_phase2' is not defined on block 0
Material property 'crysrot', requested by 'crysp_phase2' is not defined on block 0

**Comment by GiudGiud** [comment]
Hello

use Debug/show_material_props on an example input and find out which material is supposed to be declaring this

then check if you have this material in your simulation and that you are not passing parameters that would be changing the name of the property 

Guillaume

**Reply by Emmanuel-324** [reply]
So below is the .out after adding 
[Debug]
   show_material_props = true 
[]

Active Materials:
    Subdomain:  (0)
      Material Name:       elasticity_tensor_phase0
      Property Names:      "Euler_angles" "phase0_crysrot" "phase0_effective_stiffness" "phase0_elasticity_tensor" 
                             
      Material Name:       crysp_phase0
      Property Names:      "acc_slip" "fp" "gss" "lage" "phase0_Jacobian_mult" "phase0_elastic_strain" 
                             "phase0_stress" "pk2" "update_rot" 
      Material Name:       strain_phase0
      Property Names:      "phase0_deformation_gradient" "phase0_mechanical_strain" "phase0_rotation_increment" 
                             "phase0_strain_increment" "phase0_strain_rate" "phase0_total_strain" 
      Material Name:       elasticity_tensor_phase1
      Property Names:      "Euler_angles" "phase1_crysrot" "phase1_effective_stiffness" "phase1_elasticity_tensor" 
                             
      Material Name:       crysp_phase1
      Property Names:      "acc_slip" "fp" "gss" "lage" "phase1_Jacobian_mult" "phase1_elastic_strain" 
                             "phase1_stress" "pk2" "update_rot" 
      Material Name:       strain_phase1
      Property Names:      "phase1_deformation_gradient" "phase1_mechanical_strain" "phase1_rotation_increment" 
                             "phase1_strain_increment" "phase1_strain_rate" "phase1_total_strain" 
      Material Name:       elasticity_tensor_phase2
      Property Names:      "Euler_angles" "phase2_crysrot" "phase2_effective_stiffness" "phase2_elasticity_tensor" 
                             
      Material Name:       crysp_phase2
      Property Names:      "acc_slip" "fp" "gss" "lage" "phase2_Jacobian_mult" "phase2_elastic_strain" 
                             "phase2_stress" "pk2" "update_rot" 
      Material Name:       strain_phase2
      Property Names:      "phase2_deformation_gradient" "phase2_mechanical_strain" "phase2_rotation_increment" 
                             "phase2_strain_increment" "phase2_strain_rate" "phase2_total_strain" 
      Material Name:       h0
      Property Names:      "d^2h0/deta0^2" "d^2h0/deta0deta1" "d^2h0/deta0deta2" "d^2h0/deta1^2" "d^2h0/deta1deta2" 
                             "d^2h0/deta2^2" "dh0/deta0" "dh0/deta1" "dh0/deta2" "h0" 
      Material Name:       h1
      Property Names:      "d^2h1/deta0^2" "d^2h1/deta0deta1" "d^2h1/deta0deta2" "d^2h1/deta1^2" "d^2h1/deta1deta2" 
                             "d^2h1/deta2^2" "dh1/deta0" "dh1/deta1" "dh1/deta2" "h1" 
      Material Name:       h2
      Property Names:      "d^2h2/deta0^2" "d^2h2/deta0deta1" "d^2h2/deta0deta2" "d^2h2/deta1^2" "d^2h2/deta1deta2" 
                             "d^2h2/deta2^2" "dh2/deta0" "dh2/deta1" "dh2/deta2" "h2" 
      Material Name:       global_stress
      Property Names:      "Jacobian_mult" "stress" 

    Active Face Materials:
    Subdomain:  (0)
      Material Name:       elasticity_tensor_phase0_face
      Property Names:      "Euler_angles" "phase0_crysrot" "phase0_effective_stiffness" "phase0_elasticity_tensor" 
                             
      Material Name:       crysp_phase0_face
      Property Names:      "acc_slip" "fp" "gss" "lage" "phase0_Jacobian_mult" "phase0_elastic_strain" 
                             "phase0_stress" "pk2" "update_rot" 
      Material Name:       strain_phase0_face
      Property Names:      "phase0_deformation_gradient" "phase0_mechanical_strain" "phase0_rotation_increment" 
                             "phase0_strain_increment" "phase0_strain_rate" "phase0_total_strain" 
      Material Name:       elasticity_tensor_phase1_face
      Property Names:      "Euler_angles" "phase1_crysrot" "phase1_effective_stiffness" "phase1_elasticity_tensor" 
                             
      Material Name:       crysp_phase1_face
      Property Names:      "acc_slip" "fp" "gss" "lage" "phase1_Jacobian_mult" "phase1_elastic_strain" 
                             "phase1_stress" "pk2" "update_rot" 
      Material Name:       strain_phase1_face
      Property Names:      "phase1_deformation_gradient" "phase1_mechanical_strain" "phase1_rotation_increment" 
                             "phase1_strain_increment" "phase1_strain_rate" "phase1_total_strain" 
      Material Name:       elasticity_tensor_phase2_face
      Property Names:      "Euler_angles" "phase2_crysrot" "phase2_effective_stiffness" "phase2_elasticity_tensor" 
                             
      Material Name:       crysp_phase2_face
      Property Names:      "acc_slip" "fp" "gss" "lage" "phase2_Jacobian_mult" "phase2_elastic_strain" 
                             "phase2_stress" "pk2" "update_rot" 
      Material Name:       strain_phase2_face
      Property Names:      "phase2_deformation_gradient" "phase2_mechanical_strain" "phase2_rotation_increment" 
                             "phase2_strain_increment" "phase2_strain_rate" "phase2_total_strain" 
      Material Name:       h0_face
      Property Names:      "d^2h0/deta0^2" "d^2h0/deta0deta1" "d^2h0/deta0deta2" "d^2h0/deta1^2" "d^2h0/deta1deta2" 
                             "d^2h0/deta2^2" "dh0/deta0" "dh0/deta1" "dh0/deta2" "h0" 
      Material Name:       h1_face
      Property Names:      "d^2h1/deta0^2" "d^2h1/deta0deta1" "d^2h1/deta0deta2" "d^2h1/deta1^2" "d^2h1/deta1deta2" 
                             "d^2h1/deta2^2" "dh1/deta0" "dh1/deta1" "dh1/deta2" "h1" 
      Material Name:       h2_face
      Property Names:      "d^2h2/deta0^2" "d^2h2/deta0deta1" "d^2h2/deta0deta2" "d^2h2/deta1^2" "d^2h2/deta1deta2" 
                             "d^2h2/deta2^2" "dh2/deta0" "dh2/deta1" "dh2/deta2" "h2" 
      Material Name:       global_stress_face
      Property Names:      "Jacobian_mult" "stress" 

Active Neighboring Materials:
    Subdomain:  (0)
      Material Name:       elasticity_tensor_phase0_neighbor
      Property Names:      "Euler_angles" "phase0_crysrot" "phase0_effective_stiffness" "phase0_elasticity_tensor" 
                             
      Material Name:       crysp_phase0_neighbor
      Property Names:      "acc_slip" "fp" "gss" "lage" "phase0_Jacobian_mult" "phase0_elastic_strain" 
                             "phase0_stress" "pk2" "update_rot" 
      Material Name:       strain_phase0_neighbor
      Property Names:      "phase0_deformation_gradient" "phase0_mechanical_strain" "phase0_rotation_increment" 
                             "phase0_strain_increment" "phase0_strain_rate" "phase0_total_strain" 
      Material Name:       elasticity_tensor_phase1_neighbor
      Property Names:      "Euler_angles" "phase1_crysrot" "phase1_effective_stiffness" "phase1_elasticity_tensor" 
                             
      Material Name:       crysp_phase1_neighbor
      Property Names:      "acc_slip" "fp" "gss" "lage" "phase1_Jacobian_mult" "phase1_elastic_strain" 
                             "phase1_stress" "pk2" "update_rot" 
      Material Name:       strain_phase1_neighbor
      Property Names:      "phase1_deformation_gradient" "phase1_mechanical_strain" "phase1_rotation_increment" 
                             "phase1_strain_increment" "phase1_strain_rate" "phase1_total_strain" 
      Material Name:       elasticity_tensor_phase2_neighbor
      Property Names:      "Euler_angles" "phase2_crysrot" "phase2_effective_stiffness" "phase2_elasticity_tensor" 
                             
      Material Name:       crysp_phase2_neighbor
      Property Names:      "acc_slip" "fp" "gss" "lage" "phase2_Jacobian_mult" "phase2_elastic_strain" 
                             "phase2_stress" "pk2" "update_rot" 
      Material Name:       strain_phase2_neighbor
      Property Names:      "phase2_deformation_gradient" "phase2_mechanical_strain" "phase2_rotation_increment" 
                             "phase2_strain_increment" "phase2_strain_rate" "phase2_total_strain" 
      Material Name:       h0_neighbor
      Property Names:      "d^2h0/deta0^2" "d^2h0/deta0deta1" "d^2h0/deta0deta2" "d^2h0/deta1^2" "d^2h0/deta1deta2" 
                             "d^2h0/deta2^2" "dh0/deta0" "dh0/deta1" "dh0/deta2" "h0" 
      Material Name:       h1_neighbor
      Property Names:      "d^2h1/deta0^2" "d^2h1/deta0deta1" "d^2h1/deta0deta2" "d^2h1/deta1^2" "d^2h1/deta1deta2" 
                             "d^2h1/deta2^2" "dh1/deta0" "dh1/deta1" "dh1/deta2" "h1" 
      Material Name:       h2_neighbor
      Property Names:      "d^2h2/deta0^2" "d^2h2/deta0deta1" "d^2h2/deta0deta2" "d^2h2/deta1^2" "d^2h2/deta1deta2" 
                             "d^2h2/deta2^2" "dh2/deta0" "dh2/deta1" "dh2/deta2" "h2" 
      Material Name:       global_stress_neighbor
      Property Names:      "Jacobian_mult" "stress"

**Reply by GiudGiud** [reply]
hello

so the deformation_graident property name got prefixed with phase0 and phase1. Are you wanting to use a specific phase for that material property?

**Reply by Emmanuel-324** [reply]
I just got this input script from a colleague and am trying to make it work. I can share the input script if you need to see it.

**Reply by GiudGiud** [reply]
You’ll need to get context from them about this.

If they meant to use properties from a specific phase or were relying on some sort of homogenization between phases, the treatment will be different

**Reply by Emmanuel-324** [reply]
Yes, it is for a nickel-chromium alloy that exhibits anisotropy. One of the issues is the precipitates(phases) have different crystal structures
.

**Reply by Emmanuel-324** [reply]
Is there a way to use different slip systems with slip resistance in MOOSE?

**Reply by GiudGiud** [reply]
In the same spatial regions? (Subdomains)
